pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -1091,61 +1094,128 @@ func cleanupOrphanExposureClassHandlerResources(
 	log logr.Logger,
 	c client.Client,
 	exposureClassHandlers []config.ExposureClassHandler,
+	zones []string,
 ) error {
+	// Remove ordinary, orphaned istio exposure class namespaces
 	exposureClassHandlerNamespaces := &corev1.NamespaceList{}
 	if err := c.List(ctx, exposureClassHandlerNamespaces, client.MatchingLabels{v1beta1constants.GardenRole: v1beta1constants.GardenRoleExposureClassHandler}); err != nil {
 		return err
 	}
 
 	for _, namespace := range exposureClassHandlerNamespaces.Items {
-		log = log.WithValues("namespace", client.ObjectKeyFromObject(&namespace))
-
-		var exposureClassHandlerExists bool
-		for _, handler := range exposureClassHandlers {
-			if *handler.SNI.Ingress.Namespace == namespace.Name {
-				exposureClassHandlerExists = true
-				break
+		if err := cleanupOrphanIstioNamespace(ctx, log, c, namespace, true, func() bool {
+			for _, handler := range exposureClassHandlers {
+				if *handler.SNI.Ingress.Namespace == namespace.Name {
+					return true
+				}
 			}
+			return false
+		}); err != nil {
+			return err
 		}
-		if exposureClassHandlerExists {
-			continue
-		}
-		log.Info("Namespace is orphan as there is no ExposureClass handler in the gardenlet configuration anymore")
+	}
 
-		// Determine the corresponding handler name to the ExposureClass handler resources.
-		handlerName, ok := namespace.Labels[v1beta1constants.LabelExposureClassHandlerName]
-		if !ok {
-			log.Info("Cannot delete ExposureClass handler resources as the corresponging handler is unknown and it is not save to remove them")
-			continue
-		}
+	// Remove zonal, orphaned istio exposure class namespaces
+	zonalExposureClassHandlerNamespaces := &corev1.NamespaceList{}
+	labelSelector := client.MatchingLabelsSelector{
+		Selector: labels.NewSelector().Add(utils.MustNewRequirement(v1beta1constants.GardenRole, selection.Exists)).Add(utils.MustNewRequirement(v1beta1constants.LabelExposureClassHandlerName, selection.Exists)),
+	}
```

Comment:
Nit: You could inline this variable

Reply:
Inlined label selector into `List(...)` call with next commit.